### PR TITLE
[RW-321][Risk=no] Peg all dependencies to a fixed version

### PR DESCRIPTION
**Pull request by NehaBroad**

### CODE REVIEW

**Comment by NehaBroad** [comment]
apparently synk failure for security is a known issue https://github.com/spring-projects/spring-security/issues/3392

**Comment by rushtong** [comment]
I don't think this is 100% ready for review. For the snyk failures, we should make it clear in the PR what we are addressing and what we are not addressing and why. For example,
```
✗ High severity vulnerability found in org.mortbay.jetty:jetty
  Description: Arbitrary Command Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGMORTBAYJETTY-32091
  Introduced through: com.google.oauth-client:google-oauth-client-jetty@1.23.0
  From: com.google.oauth-client:google-oauth-client-jetty@1.23.0 > org.mortbay.jetty:jetty@6.1.26

✗ High severity vulnerability found in com.h2database:h2
  Description: Arbitrary Code Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-COMH2DATABASE-31685
  Introduced through: com.h2database:h2@1.4.197
  From: com.h2database:h2@1.4.197
```
Neither one has a fix ... so we can ignore them for this PR. We do need to address them eventually, but in different stories. 

Some, like the guava one:
```
✗ Medium severity vulnerability found in com.google.guava:guava
  Description: Deserialization of Untrusted Data
  Info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236
  Introduced through: com.google.cloud.sql:mysql-socket-factory@1.0.10, project :common-api@*, com.google.guava:guava@22.0, com.google.cloud:google-cloud-bigquery@0.30.0-beta, com.google.cloud:google-cloud-storage@1.8.0, com.google.truth:truth@0.42
  From: com.google.cloud.sql:mysql-socket-factory@1.0.10 > com.google.cloud.sql:jdbc-socket-factory-core@1.0.10 > com.google.guava:guava@22.0
  From: project :common-api@* > com.google.guava:guava@22.0
  From: com.google.guava:guava@22.0
  and 33 more...
```
When we do fix it, it breaks the local environment for some team members - so we've punted on that one too.

**Comment by rushtong** [comment]
Closing and re-opening is fine.

**Comment by calbach** [comment]
My 2c: I would definitely encourage an initial PR which effectively locks the versions to what we're currently using. This may generate "new" Snyk failures, but they aren't really new - they were just hidden (based on my understanding of Snyk). As a follow-up, we could fix those failures as those fixes may be more involved.

Exception: if there's some obvious low hanging fruit, e.g. a regression in the latest version of a package, and there's a very similar version one release earlier (without the issue) - you could roll that into the initial PR as well.

File: api/build.gradle [hunk 1]
```diff
@@ -239,55 +239,64 @@ dependencies {
...
-  compile('org.springframework.boot:spring-boot-starter-actuator:+') {
+  compile('org.springframework.boot:spring-boot-starter-actuator') {
     exclude module: 'spring-boot-starter-tomcat'
     exclude group: 'org.slf4j', module: 'jul-to-slf4j'
   }
-  compile 'org.springframework.security:spring-security-web:+'
+  compile('org.springframework:spring-webmvc:5.1.+') {
+    exclude module: 'spring-web'
+  }
+  compile 'org.springframework:spring-web:5.1.+'
```
Comment (rushtong) [inline api/build.gradle]: We don't want the `+`'s anywhere for this story.

**Pull request closed without merging**